dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,215 +3,151 @@
+    public async Task<string> GenerateImageAsync(
+        string description,
+        int width,
+        int height,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
     {
-        var operationId = await this.StartTextToImageAsync(description, width, height, cancellationToken).ConfigureAwait(false);
-        var result = await this.GetTextToImageResultAsync(operationId, cancellationToken).ConfigureAwait(false);
+        Verify.NotNull(description);
+        const string SizeErr = "Dall-E 2 can generate only square images of size 256x256, 512x512, or 1024x1024";
 
-        if (result.Result is null)
+        if (width != height)
         {
-            throw new KernelException("Azure OpenAI Text To Image null response");
+            throw new ArgumentOutOfRangeException("width,height", $"{width}x{height}", SizeErr);
         }
 
-        if (result.Result.Images.Count == 0)
+        var size = width switch
         {
-            throw new KernelException("Azure OpenAI Text To Image result not found");
+            256 => ImageSize.Size256x256,
+            512 => ImageSize.Size512x512,
+            1024 => ImageSize.Size1024x1024,
+            _ => throw new ArgumentOutOfRangeException(nameof(width), SizeErr)
+        };
+
+        Response<ImageGenerations> imageGenerations;
+        try
+        {
+            imageGenerations = await this._client.GetImageGenerationsAsync(
+                new ImageGenerationOptions
+                {
+                    DeploymentName = this._deploymentName,
+                    Prompt = description,
+                    Size = size,
+                }, cancellationToken).ConfigureAwait(false);
         }
-
-        return result.Result.Images.First().Url;
-    }
-
-    /// <summary>
-    /// Start an text to image task
-    /// </summary>
-    /// <param name="description">Image description</param>
-    /// <param name="width">Image width in pixels</param>
-    /// <param name="height">Image height in pixels</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns> The operationId that identifies the original text to image request. </returns>
-    private async Task<string> StartTextToImageAsync(string description, int width, int height, CancellationToken cancellationToken = default)
-    {
-        Verify.NotNull(description);
-        if (width != height || width != 256 && width != 512 && width != 1024)
+        catch (RequestFailedException e) when (e.Status == 404)
         {
-            throw new ArgumentOutOfRangeException(nameof(width), width, "OpenAI can generate only square images of size 256x256, 512x512, or 1024x1024.");
+            this._logger.LogError("Image generation failed with status code 404. This error can occur also when attempting to use Dall-E-3 which is still not supported");
```

Comment:
This error message may not be applicable with new implementation.

Reply:
Removed, the current ClientCore don't log at this stage, removing to keep it similar.